File: leetcode/70_ClimbingStairs.java
class Solution {
    public int climbStairs(int n) {
        int first = 1;
        int sec = 2;
        if( n == 1) return first;
        if( n == 2) return sec;
        int i = 3;
        while(i <= n)
        {
            int res = first + sec ;
            first = sec;
            sec = res;
            i++;
        }
        return sec;
    }
}

File: leetcode/python/70_ClimbingStairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n <= 2:
            return n 

        prev = 1 
        curr = 2
        steps = 0

        for i in range(3, n+1):
            steps = prev + curr
            prev = curr
            curr = steps
        
        return steps